packages/components/src/components/horizontal-nav/horizontal-nav.component.ts

```diff
@@ -0,0 +1,196 @@
+  }
+
+  /**
+   * Adjust the visual placement of the priority menu
+   * @param startPos The start position to calculate from
+   */
+  private adjustPriorityMenuPosition(startPos: number) {
+    const relativeX = startPos - this.priorityMenu.offsetLeft;
+    this.priorityMenu.style.transform = `translate(${relativeX}px, 0)`;
+  }
+
+  /**
+   * Determines which items should be shown or hidden, depending on the current width
+   */
+  private handlePriorityMenu() {
+    const navItems = this.getSlottedNavItems();
+    const { width } = this.navItemWrapper.getBoundingClientRect();
+
+    let firstHiddenItemRightPos: number | undefined;
+    const nextPriorityMenuItems: SynNavItem[] = [];
+
+    // eslint-disable-next-line complexity
+    navItems.forEach(item => {
+      // If we already have found an item,
+      // skip the position check as we already know the first hidden item.
+      // Else, check if the most right position of the item is in the view
+      const isHidden = firstHiddenItemRightPos || item.getBoundingClientRect().right > width;
+
+      // eslint-disable-next-line no-param-reassign
+      item.style.visibility = isHidden ? 'hidden' : 'visible';
+      item.setAttribute('aria-hidden', isHidden ? 'true' : 'false');
+
+      if (isHidden) {
+        // Save a reference to the currently hidden elements.
+        // This makes it possible to render them via renderPriorityMenu
+        nextPriorityMenuItems.push(item);
+
+        // Get the position of the first item
+        // Will get used to position the priority menu
+        if (!firstHiddenItemRightPos) {
+          firstHiddenItemRightPos = item.offsetLeft;
+        }
+      }
+    });
+
+    this.priorityMenuItems = nextPriorityMenuItems;
+
+    // Only show the priority menu label if there are no visible items anymore
+    const labelVisibility = nextPriorityMenuItems.length === navItems.length ? 'visible' : 'hidden';
+    this.priorityMenuLabelElement.style.visibility = labelVisibility;
+
+    if (firstHiddenItemRightPos) {
+      this.adjustPriorityMenuPosition(firstHiddenItemRightPos);
+    }
+  }
+
+  /**
+   * Renders the priority menu with all items that are currently not available on screen
+   */
+  private renderPriorityMenu() {
+    const items = this.priorityMenuItems;
+
+    return html`
+      <div
+        class=${classMap({
+          'priority-menu': true,
+          'priority-menu--hidden': items.length === 0,
+        })}
+        part="priority-menu"
+      >
+        <syn-nav-item>
+          <syn-icon name="more_horiz" label="More" slot="prefix"></syn-icon>
+          <span class="priority-menu__label" part="priority-menu-label">${this.priorityMenuLabel}</span>
+          <nav class="priority-menu__list" slot="children">
+            ${items.map((item, index) => html`
+              <syn-nav-item
+                ?divider=${index !== 0}
+                vertical
+              >
+                ${item.textContent}
+              </syn-nav-item>
```

Comment:
I have very big concerns in re-rendering elements here. We don't know, what people do with their nav items, maybe they use them with event listeners, maybe with links, maybe with whatever. This prevents to do anything individual with them.

I would strongly suggest to MOVE items and when moving only toggle the `vertical` attribute on syn-nav-items on the first level, but do nothing else here. I think moving between slots could be the best thing here, so we just toggle a `slot=more` attribute. 

To be honest: I think it would be even fine, if they don't use this with a `syn-nav-item` at the top – I believe as  a developer I maybe would do more something like:

```html
<syn-horizontal-nav>
<div class="nav-item-wrapper">
   <syn-nav-item>I am horizontal</syn-nav-item>
   <syn-nav-item vertical>I am vertical and do things differently</syn-nav-item>
</div>
</syn-horizontal-nav>

<styles>
.nav-item-wrapper[slot="more"] syn-nav-item:not([vertical]){display: none;}
.nav-item-wrapper:not([slot="more"]) syn-nav-item[vertical]{display: none;}
</styles>
```

Initially here the vertical nav item is hidden.

When the item doesn't have enough space, it would move: 
```html
<div class="nav-item-wrapper" slot="more">
   <syn-nav-item>I am horizontal</syn-nav-item>
   <syn-nav-item vertical>I am vertical and do things differently</syn-nav-item>
</div>
```

This would hide the horizontal element and show the vertical one.

So my proposal summarized:
1. Toggle a `slot=more` attribute to indicate if elements move into a menu or not.
2. Toggle `vertical` if those elements are `syn-nav-items`. If REALLY needed, you could even show the divider programmatically.

Reply:
Seems like a nice idea. However, we will have to make sure there are clones that occupy the space, because otherwise the navigations size will change and cause infinite rerenders. This was the reason I created copies. However as you have pointed out, even when using `cloneNode`, event listeners will not get propagated. So maybe we could do both:

1. Let the current size check stay as it is to make sure to position the dropdown correctly.
2. Create dummies in the dom with the same relative positions than the original items and set an absolute position on them with the same base coordinates (will have to be synced with the dummies then...)
3. Move the items to the dropdown completely like you suggested.

I suggest we should have a talk about this tomorrow.

Reply:
This sounds reasonable. So if I get right:
1. Invisible copies for calculating sizes.
2. Slot switching for movement.

If this somehow works, that sounds good. As an alternative: We could just store the "initial" sizes in the elements and check, if there is enough space? That's how we handled it at uni-freiburg.de